bin/cli.js

```diff
@@ -39,37 +54,6 @@ function initKnex(env, opts) {
     );
   }
 
-  if (!opts.knexfile) {
-    const configurationPath = resolveKnexFilePath();
-    const configuration = configurationPath
-      ? require(configurationPath.path)
-      : undefined;
-
-    env.configuration = configuration || mkConfigObj(opts);
-    if (!env.configuration.ext && configurationPath) {
-      env.configuration.ext = configurationPath.extension;
-    }
-  }
-  // If knexfile is specified
-  else {
-    const resolvedKnexfilePath = path.resolve(opts.knexfile);
-    const knexfileDir = path.dirname(resolvedKnexfilePath);
-    process.chdir(knexfileDir);
-    env.configuration = require(resolvedKnexfilePath);
-
-    if (!env.configuration) {
-      exit(
-        'Knexfile not found. Specify a path with --knexfile or pass --client and --connection params in commandline'
-      );
-    }
-
-    if (!env.configuration.ext) {
-      env.configuration.ext = path
-        .extname(resolvedKnexfilePath)
-        .replace('.', '');
-    }
-  }
-
```

Comment:
This logic was essentially duplicating the "configuration file discovery" logic that was already provided by Liftoff.  However, modules discovered via this control path would never utilize any of Liftoff's module pre-loaders.  This is the reason that some users were observing syntax errors when attempting to load non-JS config files via the `--knexfile` parameter.  Ex: #2998